.github/workflows/on-pr.yml

```diff
@@ -0,0 +1,110 @@
+# This workflow runs all workflows to check, build and test the project on
+# various Linux flavors, as well as on MacOS and Windows, on every push to a
+# user branch. However, it will not run if the pull request is a draft unless it
+# has the 'DraftRunCI' label.
+name: PR
+
+on:
+  push:
+    branches-ignore:
+      - develop
+      - release
+      - master
+    paths:
+      - '.github/actions/build-deps/**'
+      - '.github/actions/build-test/**'
+      - '.github/scripts/levelization/**'
+      - '.github/scripts/strategy-matrix/**'
+      - '.github/workflows/build-test.yml'
+      - '.github/workflows/check-format.yml'
+      - '.github/workflows/check-levelization.yml'
+      - '.github/workflows/notify-clio.yml'
+      - '.github/workflows/on-pr.yml'
+      # Keep the list of paths below in sync with those in `on-trigger.yml`.
```

Comment:
would it be simpler to use the reverse i.e. prefix `!` for exclusion of certain files ? https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#onpushpull_requestpull_request_targetpathspaths-ignore